account_payment_sale/models/sale_order.py

```diff
@@ -52,3 +52,17 @@ def _get_invoice_grouping_keys(self) -> list:
         if "payment_mode_id" not in keys:
             keys.append("payment_mode_id")
         return keys
+
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        momentary_mode = self.env.ref("account_payment_sale.payment_mode_momentary")
+        for order in self:
+            if not order.payment_mode_id:
+                order.payment_mode_id = momentary_mode
+
+        res = super()._create_invoices(grouped=grouped, final=final, date=date)
```

Comment:
Invoices are being created with that new payment mode, you're unassigning it from the orders but not from the invoice.